### ISSUE TITLE
Multiplication with Units (ms x deg) regression since 3.1.0

### ISSUE THREAD

**Issue by roughcoder** [sass/node-sass#1031]
After trying to port the new Material Design Lite code to a webpack project that I am working on, I stumbled upon an compile issue involving multiplication and units.

Compiling the following code works in `v3.0.0` and fails in `v3.1.0` and `v3.2.0`.

```scss
$test-variable: 360deg * 1333ms;
```

returns the following error

```
{
  "message": "",
  "column": 20,
  "line": 2,
  "file": "/Users/neilbarton/Documents/Development/project/client-app/test/app.scss",
  "status": 1
}
```

Not the most informative error message.  Am I missing some additional options from the compile command or is this a bug?

If i remove the `ms` all works good.

The offending lines within the Material Design Lite project are https://github.com/google/material-design-lite/blob/master/src/_variables.scss#L306-L314.

```
// Amount of circle the arc takes up.
$spinner-arc-size: 270deg;
// Time it takes to expand and contract arc.
$spinner-arc-time: 1333ms;
// How much the start location of the arc should rotate each time.
$spinner-arc-start-rot: 216deg;

$spinner-duration: 360deg * $spinner-arc-time / ($spinner-arc-start-rot + (360deg - $spinner-arc-size));
```

**Comment by saper** [comment]
So, the error message (hidden by node-sass, which is a bug) is:

Given `t1`:

````
$test-variable: 360deg * 1333ms ;
````

it produces:

````
Error: Incompatible units: 'ms' and 'deg'
        on line 1 of t1
>> $test-variable: 360deg * 1333ms ;
   -------------------------^
````

Your full example fails weirdly:

````
$spinner-arc-size: 270deg;
$spinner-arc-time: 1333ms;
$spinner-arc-start-rot: 216deg;

$spinner-duration: 360deg * $spinner-arc-time / ($spinner-arc-start-rot + (360deg - $spinner-arc-size));
````

The error message:

````
Error: Incompatible units: 'ms' and 'deg'
        on line 2 of t2
>> $spinner-arc-time: 1333ms;
   -------------------^
````

The support for `ms` got officially introduced in https://github.com/sass/libsass/issues/1167 so maybe previously it magically worked just because it was ignored?

**Comment by xzyfer** [comment]
This is an interesting issue.

The following is not valid in Sass.

```scss
$foo: 360deg * 1333ms;
a { b: $foo }
```
```
Error: 479880deg*ms isn't a valid CSS value.
        on line 2 of test.scss
```

LibSass is failing (correctly) but at the wrong time. Ruby Sass fails when output the css, where as LibSass fails during parsing. This is shown by @saper's first example which does not error in Ruby Sass.

The key difference here is that although `$test-variable` has an invalid value at parse time, it may later on be `*`'d or `/`'d by another value that makes it valid for output. By LiibSass failing we're prevent the value from having a chance to become valid.

An example of this in action

```
$foo: 1333ms / (216deg + 360deg - 270deg); // Ruby Sass throws an error
a { b: $bar }
```
```
$foo: 360deg *  1333ms / (216deg + 360deg - 270deg); // *'ing by deg makes the value valid
a { b: $bar }
```

**Comment by xzyfer** [comment]
This is fixed in 3.4.0-beta1.

**Comment by xzyfer** [comment]
Fixed in https://github.com/sass/node-sass/pull/1214. Will be in v3.4.0.
